Apps/GatewayApi/test/unit/Services.Test/UserValidationServiceTests.cs

```diff
@@ -0,0 +1,170 @@
+// Copyright © 2019 Province of British Columbia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-------------------------------------------------------------------------
+namespace HealthGateway.GatewayApiTests.Services.Test
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using HealthGateway.Common.Constants;
+    using HealthGateway.GatewayApi.Models;
+    using HealthGateway.GatewayApi.Services;
+    using Microsoft.Extensions.Configuration;
+    using Moq;
+    using Xunit;
+
+    /// <summary>
+    /// UserValidationService's Unit Tests.
+    /// </summary>
+    public class UserValidationServiceTests
+    {
+        private const string Hdid = "hdid-mock";
+
+        /// <summary>
+        /// IsPhoneNumberValidAsync returns false for invalid phone number.
+        /// </summary>
+        /// <param name="phoneNumber">The phone number to validate.</param>
+        /// <param name="expected">The expected value of the validation result.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [InlineData("3345678901", true)]
+        [InlineData("2507001000", true)]
+        [InlineData("xxx3277465", false)]
+        [InlineData("abc", false)]
+        [Theory]
+        public async Task PhoneNumberIsNotValid(string phoneNumber, bool expected)
+        {
+            // Arrange
+            PhoneNumberValidMock mock = SetupPhoneNumberValidMock(phoneNumber);
+
+            // Act
+            bool actual = await mock.Service.IsPhoneNumberValidAsync(mock.PhoneNumber);
+
+            // Assert
+            Assert.Equal(expected, actual);
+        }
+
+        /// <summary>
+        /// ValidateEligibilityAsync.
+        /// </summary>
+        /// <param name="minPatientAge">The minimum patient age to validate against.</param>
+        /// <param name="patientAge">The patient age to validate.</param>
+        /// <param name="expected">The expected eligibility result.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [InlineData(0, 0, true)]
+        [InlineData(19, 19, true)]
+        [InlineData(19, 20, true)]
+        [InlineData(19, 18, false)]
+        [Theory]
+        public async Task ShouldValidateEligibilityAsync(int minPatientAge, int patientAge, bool expected)
```

Comment:
```suggestion
        public async Task ShouldValidateEligibility(int minPatientAge, int patientAge, bool expected)
```

Nit: I don't think we need to append "Async" to the names of methods in our unit tests, even though that's what we do in the rest of the project.  We don't really need to differentiate between async and non-async versions of a test, because we will only ever have one or the other.

Note that a unit test might be named with a reference to another method that has an Async suffix, and that wouldn't be extraneous because it helps us identify the specific method.

For example:

```
ValidateEligibilityAsyncShouldThrowException()
```
^ This is fine.

```
ValidateEligibilityAsyncShouldThrowExceptionAsync()
```
^ This is extra.

As a further example, with a method named `CreateProfileAsync`, a test could be named `ShouldCreateProfile` or `ShouldCreateProfileAsync` and be fine either way, though I would lean to the former as it describes expected behaviour.